statemachines/pre_flight_checks/test_hostname_resolves

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+DIR=`dirname $0`
+. /etc/restatemachine/shell_helpers.inc
+
+INPUT=`cat`
+Error () {
+        echo -n "$INPUT"
+        restatemachine_transition_to "test_hostname_resolves"
+        restatemachine_transition_after_seconds "60"
+        restatemachine_status_message "$1"
+        exit 0
+}
+
+HOSTNAME=`echo "$INPUT" | jq -r '.hostname'`
+
+RESOLVES=`dig "$HOSTNAME" @8.8.8.8 | grep -q 'NXDOMAIN'`
```

Comment:
There are other answers than NXDOMAIN that is also invalid, we should instead check for a positive number in the answer section and NOERROR as status.

Example of things not caught by the current version:
SERVFAIL
timeouts
etc

Better version:
if ! dig "$HOSTNAME" @8.8.8.8 | tr "\n" " " | grep -q 'status:[[:space:]]*NOERROR.*[[:space:]]ANSWER:[[:space:]]*[1-9]';  then
# error goes here